Remove paths part size hash from Matcher

matthewmcgarvey opened the pull request:
## Summary

Refer to https://github.com/luckyframework/lucky_router/issues/27 for a more detailed explanation of why this change is being made, but TLDR is that this speeds up the case of finding a matching request without sacrificing behavior.

This benchmark update is to better compare the code when it is warmed up. It runs the router much more but fewer times.

## Benchmark (with `--release`)

- Before: 224 ms
- After: 212 ms
- Difference: 12 ms
- Improvement: 5%

paulcsmith commented:
Will take a look at this soon! One thing to note is that I think those benchmarks are without the `--release` flag. Sometimes in release mode things can change a lot. So I think it'd be good to check the previous merged change and this one with `--release` to make sure the numbers really are better. Maybe they're even bigger than we previously thought!

matthewmcgarvey commented:
@paulcsmith Good call! I updated the benchmark and re-ran with release flag and am seeing very different results. I updated the description with the results.

matthewmcgarvey commented:
@paulcsmith Is this ready to be merged?
